src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -300,9 +379,17 @@ func (s *commitLogSource) ReadData(
 			blockStart: dp.Timestamp.Truncate(blockSize),
 		}
 	}
+	doneReadingCommitlogs()
```

Comment:
Is it a waste to just emit the gauges the whole time this function is running?  Then we could do it all at the start and defer all these cleanups.  It would be better mainly because it would defend against an early return not stopping the loop which would cause a memory leak (for loop continuing to hold ref to the commit log source forever.

Reply:
see comment below